Source/WebCore/platform/graphics/mac/controls/ControlMac.mm

```diff
@@ -176,6 +177,43 @@
     [WebControlWindow setHasKeyAppearance:style.states.contains(ControlStyle::State::WindowActive)];
 }
 
+static void drawViewlessCell(const FloatRect& rect, float deviceScaleFactor, const ControlStyle& style, NSCell *cell, bool flipContext)
+{
+    CGContextRef cgContext = [[NSGraphicsContext currentContext] CGContext];
+    CGContextStateSaver stateSaver(cgContext);
+
+    [cell _setFallbackBackingScaleFactor:deviceScaleFactor];
+
+#if USE(NSVIEW_SEMANTICCONTEXT)
+    if (style.states.contains(ControlStyle::State::FormSemanticContext))
+        [cell _setFallbackSemanticContext:NSViewSemanticContextForm];
+#endif
+
+    auto adjustedRect = rect;
+
+    // FIXME: Remove flipping the coordinates if rdar://105194487 is fixed.
+    if (flipContext) {
+        // Move the coordinates origin to topleft of rect.
+        CGContextTranslateCTM(cgContext, adjustedRect.x(), adjustedRect.y());
+        adjustedRect.setLocation(FloatPoint::zero());
+
+        // Flip the coordinates.
+        CGContextTranslateCTM(cgContext, 0, adjustedRect.height());
+        CGContextScaleCTM(cgContext, 1, -1);
+    }
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wnonnull"
```

Comment:
We should probably ask AppKit for a version of this method which that doesn't require a view, so that we don't need these.

Reply:
I added a FIXME comment and I filed rdar://105250010.